### PR TITLE
Add multisig as a new scripttype enum

### PR DESCRIPTION
**Pull request by kderme**

### CODE REVIEW

**Review by sorki — DISMISSED** [review]
LGTM! Thanks for adding this.

File: cardano-db-sync/src/Cardano/DbSync/Era/Shelley/Generic/Tx.hs [hunk 1]
```diff
@@ -139,11 +140,14 @@ data TxRedeemer = TxRedeemer
 
 data TxScript = TxScript
   { txScriptHash :: !ByteString
+  , txScriptType :: ScriptType
   , txScriptPlutusSize :: Maybe Word64
   , txScriptJson :: Maybe ByteString
   , txScriptCBOR :: Maybe ByteString
   }
 
+data ScriptType = MultiSig | Timelock | Plutus
```
Comment (erikd) [inline cardano-db-sync/src/Cardano/DbSync/Era/Shelley/Generic/Tx.hs:149]: Why not just use the `ScriptType` defined in `cardano-db/src/Cardano/Db/Types.hs` ?

File: cardano-db-sync/src/Cardano/DbSync/Era/Shelley/Insert.hs [hunk 2]
```diff
@@ -888,9 +888,10 @@ insertScript tracer txId script = do
   where
     scriptType :: DB.ScriptType
     scriptType =
-      case Generic.txScriptPlutusSize script of
-        Nothing -> DB.Timelock
-        Just _ -> DB.Plutus
+      case Generic.txScriptType script of
+        Generic.MultiSig -> DB.MultiSig
+        Generic.Timelock -> DB.Timelock
+        Generic.Plutus -> DB.Plutus
```
Comment (erikd) [inline cardano-db-sync/src/Cardano/DbSync/Era/Shelley/Insert.hs:894]: Then we would not need this.

File: cardano-db-sync/src/Cardano/DbSync/Era/Shelley/Generic/Tx.hs (hunk 1 above)
Comment (kderme) [reply cardano-db-sync/src/Cardano/DbSync/Era/Shelley/Generic/Tx.hs:149]: We don't import anything from Cardano.Db in the Generic module, so I thought I'd keep it that way.

Comment (erikd) [reply cardano-db-sync/src/Cardano/DbSync/Era/Shelley/Generic/Tx.hs:149]: `cardano-db-sync/src/Cardano/DbSync/Era/Shelley/Generic/Util.hs` has
```
import qualified Cardano.Db as Db
```

**Review by erikd — APPROVED** [review]

**Pull request merged**